pkg/kube/inject/webhook.go

```diff
@@ -985,9 +991,17 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 
 	proxyConfig := wh.env.GetProxyConfigOrDefault(pod.Namespace, pod.Labels, pod.Annotations, wh.meshConfig)
 	deploy, typeMeta := kube.GetDeployMetaFromPod(&pod)
+
+	var podNamespace *corev1.Namespace
+	if wh.kubeClient != nil {
```

Comment:
I think I tried something similar once and there was concern of api calls to apiserver. I'll leave this up to the networking maintainers.

Reply:
This is actually using an informer under the hood but we need to make one and use it everywhere rather than create one per request. Also need to check it's synced.

Can we disable this when not on open shift somehow?